src/Type.h

```diff
@@ -398,6 +398,16 @@ class TableType : public IndexType
 public:
 	TableType(TypeListPtr ind, TypePtr yield);
 
+	// Used by script compilation to update a "stub" table type
+	// (which is specified by using a nil "ind" value in the constructor)
+	// with its actual index & yield - necessary for dealing with
+	// recursive types.
+	void SetIndexAndYield(TypeListPtr ind, TypePtr yield)
```

Comment:
I suspect this is not safe to be used more than once and/or only in specific circumstances otherwise current table content and index/yield types are inconsistent. Could we runtime protect from wrong usage?


Pondering a bit, we're adding this to the public API, and even though it says "used by script compilation", it is provided to anyone for use until we deprecate it. I could also see friending the CPP script_opt implementation to avoid adding it for anyone. I might be overthinking this...

Reply:
An alternative could be to have a separate CPPTableType version inherited from TableType (or IndexType). The member variables themselves are already protected, so could fiddle with them that way.

Reply:
I like the `CPPTableType` approach, and have implemented that.  Will push once we resolve the above re error message canonicalization.

Reply:
Now pushed.